src/main/java/duke/Task.java

```diff
@@ -0,0 +1,40 @@
+package duke;
+
+/**
+ * Abstract class to represent a task.
+ * Contains task name and completed status.
+ *
+ * @author YanCheng
+ */
+
+public abstract class Task {
+    protected String taskName;
+    protected boolean isDone;
+
+    /**
+     * Constructor for Task.
+     * @param taskName Name of task
+     */
+    public Task(String taskName) {
+        this.taskName = taskName;
+        isDone = false;
+    }
+
+    /**

```

Comment:
I like how you added JavaDoc comments in every public method :) makes for very comprehensive code.